pkg/script/script.go

```diff
@@ -53,6 +54,9 @@ func BuildBootstrapScript(bootstrapScripts []string, cluster *cluster.Cluster) (
 		}
 		userData = append(userData, scriptData...)
 	}
+	if len(userData) > 16384 {
```

Comment:
+1 to the idea of validating it :100:

> I'm wondering if it would make sense to gzip the user data, which will give a bit more flexibility when it comes to the limit.

That could be a good feature :slightly_smiling_face: 

> I'm also wondering if there is a way to determine which cloud provider the user data is intended for here, since the limits vary between different cloud providers.
> 
> @xmudrii thoughts?

AFAIK, only AWS has such a limit, so it would be nice to introduce the validation only for AWS. I'm not sure, but you might give it a try to find what's the cloud provider from the cluster object. I can't remember do we store that information somewhere there though.

Reply:
There is no need to decompress the file, cloud-init will automatically decompress if needed.

Here is how we are handling the compression in cluster-api-provider-aws: https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/e5e696c25960757aed0f170f41aef0b09035bb73/pkg/cloud/services/ec2/instances.go#L273-L291

Reply:
I removed the check for the 16kb limit because the script would need to be something like 18mb large to exceed the limit after it is compressed